src/crudodb.ts

```diff
@@ -0,0 +1,210 @@
+import { InternalStoreEntry, StoreSchema } from './store-schema';
+import { Database } from './database';
+import { CrudApi } from './crud-api';
+import {
+  evaluateDbVersion,
+  execDatabase,
+  generateTempKey,
+  initGeneralDb,
+  prepareStoreWithDatabase
+} from './utils';
+
+interface RegisterSchemaArgs<T> {
+  schema: StoreSchema;
+  api?: CrudApi<T>;
+  schemaKey?: string;
+}
+
+export class CrudoDb {
+
+  static async setup(debug: boolean = false): Promise<CrudoDb> {
+    const instance = new CrudoDb(debug);
+    if (debug) {
+      console.time('CrudoDb initialized');
+    }
+    return instance
+      .setup()
+      .then(() => instance)
+      .finally(() => debug && console.timeEnd('CrudoDb initialized'));
+  }
+
+  private databaseSchemas: Record<string, StoreSchema> = {};
+
+  private databases: Record<string, Database<unknown>> = {};
+
+  private get general(): Database<InternalStoreEntry> {
+    return this.databases.general as Database<InternalStoreEntry>;
+  }
+
+  private idbDatabases: Record<string, IDBDatabase> = {};
+
+  private constructor(private debug: boolean = false) {}
+
+  public close(): void {
+    Object.values(this.idbDatabases).forEach((db) => db.close());
+  }
+
+  public async get<T>(
+    schemaKey: string,
+    id: T[keyof T]
+  ): Promise<T | undefined> {
+    return execDatabase(
+      schemaKey,
+      this.databases[schemaKey],
+      (db: Database<T>) => db.get(id)
+    );
```

Comment:
The helper used requires more lines than doing the validation explicitly:

```suggestion
    validateSchemaKey(schemaKey);
    return this.databases[schemaKey].get(id);
```

This applies to all of these functions.